pkg/apis/core/v1/conversion.go

```diff
@@ -327,6 +329,35 @@ func Convert_v1_Pod_To_core_Pod(in *v1.Pod, out *core.Pod, s conversion.Scope) e
 	return nil
 }
 
+func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatus, s conversion.Scope) error {
+	autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s)
+	// autoconvert sets the core PodIPs to the v1 PodIPs
```

Comment:
Chatted about this with you earlier, but the autoconvert function is going to overwrite the existing core podIPs, so you'll lose that history. Probably need to save that array before calling autoconvert.